### DISCUSSION TITLE
Performance

### DISCUSSION THREAD

**Discussion by xu4wang** [directus/directus#11563] [Archive]
Dear All

Did anybody make some performance test with directus?

Especially, given a typical configuration (host machine + database server in one machine).
1) How many QPS directus can support?
2) How many concurrent user can access the APP?

Or any directions on improve the performance:
1) Load balance configuration
2) Parameters adjustment  
...

Thanks,Austin

**Comment by rijkvanzanten** [comment]
The answer to those first questions really depends on what database vendor you use, what type of data you request, whether or not you're gonna use authenticated users, whether or not you have caching enabled etc etc etc. I don't think there's going to be a realistic answer to the first two questions.

That being said, I do highly recommend horizontally scaling your Directus instance if you're planning on running it at scale. Make sure you use Redis for caches / rate limiter, and S3 or another shared file storage for the file storage. At that point, the bottleneck will become the amount of allowed connections and the overall server performance of the database. That being said, there's a lot of database services nowadays that scale virtually endless, like Amazon Aurora or CockroachDB

**Reply by xu4wang** [reply]
Thanks for the suggestion.

By "horizontally scaling your Directus instance", do you mean setting up multiple directus instances connecting to one central database? 

Are there any document on the setup?  Or there is no need for document since it's naturally supported, as the session/token are kept in the database and there is no state information stored in the directus instance?

**Reply by casualmatt** [reply]
You can find some install guide here, [my favorite is on AWS](https://docs.directus.io/getting-started/installation/aws/) or using CockroachDB + hetzner cloud. 

Auth is based on JWT so it's scalable on multiple servers, no special setup.

**Reply by rijkvanzanten** [reply]
> Thanks for the suggestion.
> 
> By "horizontally scaling your Directus instance", do you mean setting up multiple directus instances connecting to one central database?
> 
> Are there any document on the setup? Or there is no need for document since it's naturally supported, as the session/token are kept in the database and there is no state information stored in the directus instance?

That's exactly it! And yes, as long as you make sure you rely on a shared solution for cache/rate-limit/database/storage, it will just work 👍🏻

**Reply by xu4wang** [reply]
ha...great!
